data/packagekit.service.in

```diff
@@ -10,3 +10,4 @@ Type=dbus
 BusName=org.freedesktop.PackageKit
 User=@PACKAGEKIT_USER@
 ExecStart=@libexecdir@/packagekitd
+Wants=network-online.target
```

Comment:
OOI, if that is somehow causing PackageKit to not think its offline, it is a *very* subtle mechanism which should be explained.

As this does not add `After=network-online.target` this does not change the boot ordering *at all*. Also, being D-Bus activated, packagekit usually does not start during the boot phase in the first place, but on demand in a desktop session. Then the network is usually already running. But you mentioned "does not slow down boot", so I assume you are aware of that, and this was your intention.

So the only thing that this does is to call NetworkManager-wait-online, aka `nm-online -s -q`. So if that has *any* effect on NetworkManager's [Connectivity property](https://developer-old.gnome.org/NetworkManager/unstable/gdbus-org.freedesktop.NetworkManager.html#gdbus-property-org-freedesktop-NetworkManager.Connectivity), I'd call that a bug in NM. The property should be completely independent from whether someone called `nm-online`.

network-online.target is a bad and sloppy hack for broken legacy software, and normally only useful to change the *ordering* of the boot sequence. Merely activating it without waiting for it is *supposed* to be a no-op.

So does this exploit some subtle side effect of calling `nm-online`? Or is it really just a no-op?